### ISSUE TITLE
 Add public Architecture enum value for s390x

### ISSUE THREAD

**Issue by uweigand** [dotnet/runtime#52909]
## Background and Motivation

We are in the processes of adding support for Linux on the IBM Z architecture (s390x-ibm-linux) to .NET.   As part of this effort, the `System.Runtime.InteropServices.Architecture` enum needs to be extended by a new value to refer to this architecture.

## Proposed API

```diff
namespace System.Runtime.InteropServices
{
    public enum Architecture
    {
        X86,
        X64,
        Arm,
        Arm64,
        Wasm,
+       S390x,
    }
}

```

Full implementation is provided here: https://github.com/dotnet/runtime/pull/52906

## Usage Examples

Code can use the existing properties `System.Runtime.InteropServices.OSArchitecture` and `System.Runtime.InteropServices.ProcessArchitecture`, which will now return the new value when running on the IBM Z architecture.

## Alternative Designs

n/a

## Risks

There may be existing code that uses `System.Runtime.InteropServices.OSArchitecture` and `System.Runtime.InteropServices.ProcessArchitecture` but does not expect any new return value and will fail if it is returned.  I've seen one example of such code in vstest.  Such code would have to be updated.

**Comment by akoeplinger** [comment]
@terrajobst this should be a quick one for API review.

**Comment by iSazonov** [comment]
Also there is System.Reflection.ProcessorArchitecture.

**Comment by bartonjs** [comment]
Looks good as proposed

```diff
namespace System.Runtime.InteropServices
{
    public enum Architecture
    {
        X86,
        X64,
        Arm,
        Arm64,
        Wasm,
+       S390x,
    }
}
```

**Comment by uweigand** [comment]
> Also there is System.Reflection.ProcessorArchitecture.

This is related to the assembly file format, right?   We currently don't have a processor-specific file format since we don't (yet) support AOT.   Once we add support for that, I guess we'd have to allocate a new System.Reflection.ProcessorArchitecture (as well as System.Reflection.ImageFileMachine I think).   I'll create a new API issue for those when we get there.

**Comment by jkotas** [comment]
> System.Reflection.ProcessorArchitecture.

It is for architecture specific assemblies that are really only a thing on Windows. AOT should not require it.

**Comment by jkotas** [comment]
Fixed by #52906

**Comment by uweigand** [comment]
> > System.Reflection.ProcessorArchitecture.
> 
> It is for architecture specific assemblies that are really only a thing on Windows. AOT should not require it.

Thanks for the clarification!
